prusti-specs/src/specifications/preparser.rs

```diff
@@ -289,557 +58,547 @@ impl Parse for SpecEntArgs {
                     }
+
+                    trigger_set = TriggerSet(vec_of_triggers);
                 }
-                (match kind {
-                    Precondition => &mut pres,
-                    Postcondition => &mut posts,
-                    _ => unreachable!()
-                }).push(conjunct);
+                Err(err) => return Err(self.error_expected_tuple(err.span()))
             }
-
-            let conjunct = AssertionWithoutId {
-                kind: Box::new(common::AssertionKind::SpecEntailment {
-                    closure: lhs,
-                    arg_binders: SpecEntailmentVars {
-                        spec_id: common::SpecificationId::dummy(),
-                        pre_id: (),
-                        post_id: (),
-                        args: vars,
-                        result: Arg { name: syn::Ident::new("result", Span::call_site()),
-                                      typ: syn::parse2(quote! { i32 }).unwrap() },
-                    },
-                    pres,
-                    posts,
-                })
-            };
-
-            self.conjuncts.push(conjunct);
-            self.previous_expression_resolved = true;
-            self.expected_only_operator = true;
-            self.expected_operator = true;
-            return Ok(());
-        } else {
-            return Err(self.error_expected_bracket());
         }
+
+        Ok(AssertionWithoutId {
+            kind: box common::AssertionKind::ForAll(
+                ForAllVars {
+                    spec_id: common::SpecificationId::dummy(),
+                    id: (),
+                    vars
+                },
+                trigger_set,
+                body,
+            )
+        })
     }
-    fn resolve_parenthesized_block(&mut self, group: Group) -> syn::Result<()>{
-        // handling a parenthesized block
-        if self.expected_only_operator {
-            return Err(self.error_expected_operator());
-        }
 
-        if self.expr.is_empty() && (self.input.peek_any_operator() || self.input.is_empty()) {
-            // if there is no expression currently being parsed (in which case the
-            // parenthesized block would be a part of it) and there is a Prusti operator
-            // or nothing at all after the parenthesized block, then we might parser
-            // this as a Prusti assertion
+    fn parse_rust_until(&mut self, terminator: &str) -> syn::Result<ExpressionWithoutId> {
+        let mut t = vec![];
 
-            if self.expected_operator {
-                return Err(self.error_expected_operator());
-            }
+        while !self.peek_operator("|=") &&
+              !self.peek_operator("&&") &&
+              !self.peek_operator("==>") &&
+              !self.peek_operator(terminator) &&
+              !self.tokens.is_empty() {
+            t.push(self.pop().unwrap());
+        }
+        let mut stream = TokenStream::new();
+        stream.extend(t.into_iter());
```

Comment:
I think you can do
```suggestion
        let mut stream = TokenStream::from(t.into_iter());
```

Reply:
I don't think I can do that - the compiler complains:
```
the trait `From<std::vec::IntoIter<TokenTree>>` is not implemented for `TokenStream2`
```